packages/contracts/contracts/engine/V3/GenArt721CoreV3_Engine.sol

```diff
@@ -353,63 +383,61 @@ contract GenArt721CoreV3_Engine is
     }
 
     /**
-     * @notice Initializes contract.
-     * @param _tokenName Name of token.
-     * @param _tokenSymbol Token symbol.
-     * @param _randomizerContract Randomizer contract.
+     * @notice Initializes the contract with the provided `engineConfiguration`.
+     * This function should be called atomically, immediately after deployment.
+     * Only callable once. Validation on `engineConfiguration` is performed by caller.
+     * @param engineConfiguration EngineConfiguration to configure the contract with.
      * @param _adminACLContract Address of admin access control contract, to be
      * set as contract owner.
-     * @param _startingProjectId The initial next project ID.
-     * @param _autoApproveArtistSplitProposals Whether or not to always
-     * auto-approve proposed artist split updates.
-     * @param _splitProviderAddress Address to use as royalty splitter provider for the contract.
-     * @param _nullPlatformProvider Enforce always setting zero platform provider fees and addresses.
-     * @param _allowArtistProjectActivation Allow artist to activate their own projects.
-     * @dev _startingProjectId should be set to a value much, much less than
-     * max(uint248), but an explicit input type of `uint248` is used as it is
-     * safer to cast up to `uint256` than it is to cast down for the purposes
-     * of setting `_nextProjectId`.
-     */
-    constructor(
-        string memory _tokenName,
-        string memory _tokenSymbol,
-        address _renderProviderAddress,
-        address _platformProviderAddress,
-        address _randomizerContract,
-        address _adminACLContract,
-        uint248 _startingProjectId,
-        bool _autoApproveArtistSplitProposals,
-        address _splitProviderAddress,
-        bool _nullPlatformProvider,
-        bool _allowArtistProjectActivation
-    ) ERC721_PackedHashSeed(_tokenName, _tokenSymbol) {
-        _onlyNonZeroAddress(_renderProviderAddress);
-        // @dev checks on platform provider addresses performed in _updateProviderSalesAddresses
-        if (_nullPlatformProvider) {
-            // set platform to zero revenue splits
-            _platformProviderPrimarySalesPercentage = 0;
-            defaultPlatformProviderSecondarySalesBPS = 0;
+     */
+    function initialize(
+        EngineConfiguration calldata engineConfiguration,
+        address _adminACLContract
+    ) external {
+        // can only be initialized once
+        if (initialized) {
```

Comment:
nit: would be good to test if-branch here to ensure test failure if the check were removed

Reply:
good call 🙏 Added tests to `GenArt721CoreV3_ContractConfigure.test` to make sure re-initialization is not possible 🙏